sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -703,14 +678,11 @@ def execute_item_batch(
         """
         request_options = build_options(kwargs)
         request_options["partitionKey"] = self._set_partition_key(partition_key)
-        response_hook = kwargs.pop('response_hook', None)
         request_options["disableAutomaticIdGeneration"] = True
 
         result = self.client_connection.Batch(
             collection_link=self.container_link, batch_operations=batch_operations, options=request_options, **kwargs)
-        if response_hook:
-            response_hook(self.client_connection.last_response_headers, result)
-        return result
+        return cast(List[Dict[str, Any]], result)
```

Comment:
This cast shouldn't be necessary as `Batch` is typed as returning `List[Dict[str, Any]]`